File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
private:
int firstoccurence(vector<int>nums,int target){
    int n=nums.size();
    int start=0;
    int end=n-1;
    int ans=-1;
    while(start<=end){
        int mid=start+(end-start)/2;
        if(nums[mid]==target){
            ans=mid;
            end=mid-1;
        }
        else if(nums[mid]<target){
            start=mid+1;
        }
        else{
            end=mid-1;
        }
    }
    return ans;
}
int lastoccurence(vector<int>&nums,int target){
    int n=nums.size();
    int start=0;
    int end=n-1;
    int ans=-1;
    while(start<=end){
        int mid=start+(end-start)/2;
        if(nums[mid]==target){
            ans=mid;
            start=mid+1;
        }
        else if(nums[mid]<target){
            start=mid+1;
        }
        else{
            end=mid-1;
        }
    }
    return ans;

}
public:
    vector<int> searchRange(vector<int>& nums, int target) {
      int first=firstoccurence(nums,target);
      int last=lastoccurence(nums,target);
      return{first,last};
        
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
      int first_pos=-1;
      int last_pos=-1;
      for(int i=0;i<nums.length;i++){
        if(nums[i]==target){
            if(first_pos==-1){
                first_pos=i;
            }
            last_pos=i;
        }
      }
      return new int[]{first_pos,last_pos};
    }
}